firrtl/src/main/scala/firrtl/transforms/Dedup.scala

```diff
@@ -9,3 +9,8 @@ import firrtl.annotations._
 case class NoDedupAnnotation(target: ModuleTarget) extends SingleTargetAnnotation[ModuleTarget] {
   def duplicate(n: ModuleTarget): NoDedupAnnotation = NoDedupAnnotation(n)
 }
+
+/** Assign the targeted module to a dedup group. Only modules in the same group may be deduplicated. */
+case class DedupGroupAnnotation(target: ModuleTarget, group: String) extends SingleTargetAnnotation[ModuleTarget] {
+  def duplicate(n: ModuleTarget): DedupGroupAnnotation = DedupGroupAnnotation(n, group)
+}
```

Comment:
Is this how we want to expose this or should we try changing Chisel to just emit this annotation for all Modules based on the `desiredName` of the class?

Reply:
I think typically we want a higher-level API than manually finding modules and group them via an aspect. I will grant that the typical use will probably want more information than just desiredName, but we should probably be integrating this information in semantically useful ways, eg. a first class prefixing API which is the stand-in for hierarchical location, and automatically generating desiredName from the parameters as a way of integrating those, etc. In any case, if we want this to be an API the user can call directly, it should be a function that takes a module, not the annotation directly since we are moving away from annotations in general.

Reply:
agree that the function is what we should expose not the annotation.

I'm fine with making this a user API for now vs assuming its the desiredName that would be the grouping mechanism

Reply:
Alright, I think this is worth merging and we can come up with nicer default behavior in the future. It's useful enough as is.